test/200_fullchain-bot.test.ts

```diff
@@ -0,0 +1,287 @@
+            const initialScores = await getInitialScores(workers);
+            const contributions = tasksAndWorkers[taskIndex];
+            for (const contribution of contributions) {
+                const { worker, useEnclave, result } = contribution;
+                const { resultDigest } = buildUtf8ResultAndDigest(result);
+
+                if (useEnclave) {
+                    await iexecWrapper.contributeToTeeTask(dealId, taskIndex, resultDigest, worker);
+                } else {
+                    await iexecWrapper.contributeToTask(dealId, taskIndex, resultDigest, worker);
+                }
+            }
+            // Reveal contributions for all workers
+            for (const contribution of contributions) {
+                const { worker, result } = contribution;
+                const { resultDigest } = buildUtf8ResultAndDigest(result);
+                if (result !== '<timeout reached>') {
+                    await iexecPoco
+                        .connect(worker)
+                        .reveal(taskId, resultDigest)
+                        .then((tx) => tx.wait());
+                }
+            }
+            const { results } = buildUtf8ResultAndDigest(tasksAndWorkers[taskIndex][0].result);
+            const finalizeTx = await iexecPoco.connect(scheduler).finalize(taskId, results, '0x');
+            await finalizeTx.wait();
+            expect((await iexecPoco.viewTask(taskId)).status).to.equal(TaskStatusEnum.COMPLETED);
+            // Multiply amount by the number of finalized tasks to correctly compute
+            // stake and reward amounts.
+            const completedTasks = taskIndex + 1;
+            // Verify token balance changes
+            const expectedProxyBalanceChange = -(
+                taskPrice * completedTasks +
+                schedulerStakePerTask * completedTasks
+            );
+            const winningWorkers = contributions
+                .filter(
+                    (contribution) => contribution.result === 'iExec BOT ' + taskIndex.toString(),
+                )
+                .map((contribution) => contribution.worker);
+            const loosingWorkers = contributions
+                .filter(
+                    (contribution) => contribution.result !== 'iExec BOT ' + taskIndex.toString(),
+                )
+                .map((contribution) => contribution.worker);
+            const nonParticipantWorkers = workers.filter(
+                (worker) => !winningWorkers.includes(worker) && !loosingWorkers.includes(worker),
+            );
+            const expectedWorkerBalanceChange =
+                workerStakePerTask + workerRewardPerTask / winningWorkers.length;
+            expect(finalizeTx).to.changeTokenBalances(
+                iexecPoco,
+                [proxyAddress, ...accounts],
+                [
+                    expectedProxyBalanceChange, // Proxy
+                    -taskPrice * completedTasks, // Requester
```

Comment:
I'm surprised of this one, why does it depend on `completedTasks` and not the others (scheduler, app, ..)?

Reply:
Hum let me check